src/snowflake/snowpark/_internal/analyzer/snowflake_plan_node.py

```diff
@@ -58,6 +106,17 @@ def __init__(
         self.data = data
         self.schema_query = schema_query
 
+    @property
+    def individual_complexity_stat(self) -> Counter[str]:
+        # select $1, ..., $m FROM VALUES (r11, r12, ..., r1m), (rn1, ...., rnm)
+        # TODO: use ARRAY_BIND_THRESHOLD
+        return Counter(
```

Comment:
ah then the complexity of create_dataframe will be very high compared with select from a table. If the table is large, then at least execution of selecting from a table will be more "complex"?

Reply:
that's right. even if data created from `create_dataframe` may be smaller than a table and thus be less expensive in terms of execution, this complexity is only concerned with compiler complexity since we won't be able to estimate execution complexity without more data.

Reply:
@sfc-gh-aalam i think @sfc-gh-jdu got a good point here, and i recall we typically just use variable binding for this, and if data is coming from client side, i assume it could be small, maybe we can simply count the column, and have a VALUE category, and set 1 for value